### ISSUE TITLE
App crashing on start-up

### ISSUE THREAD

**Issue by matt4682** [KieronQuinn/PixelLauncherMods#99]
I just updated my Pixel 7 Pro to the March 2023 Pixel update (`TQ2A.230305.008.A3`) and noticed my custom icons didn't apply. Opening the app to manually apply them causes the app to immediately crash.

Here's the relevant logcat:

```
03-18 14:24:54.983 15992 16019 E AndroidRuntime: FATAL EXCEPTION: DefaultDispatcher-worker-6
03-18 14:24:54.983 15992 16019 E AndroidRuntime: Process: com.kieronquinn.app.pixellaunchermods, PID: 15992
03-18 14:24:54.983 15992 16019 E AndroidRuntime: java.lang.RuntimeException: Buffer not large enough for pixels
03-18 14:24:54.983 15992 16019 E AndroidRuntime: 	at android.graphics.Bitmap.copyPixelsFromBuffer(Bitmap.java:655)
03-18 14:24:54.983 15992 16019 E AndroidRuntime: 	at com.kieronquinn.app.pixellaunchermods.utils.extensions.Extensions_BitmapKt.Bitmap_decodeRawBitmap(Extensions+Bitmap.kt:13)
03-18 14:24:54.983 15992 16019 E AndroidRuntime: 	at com.kieronquinn.app.pixellaunchermods.repositories.IconLoaderRepositoryImpl.resizeThemedIcon(IconLoaderRepository.kt:450)
03-18 14:24:54.983 15992 16019 E AndroidRuntime: 	at com.kieronquinn.app.pixellaunchermods.repositories.RemoteAppsRepositoryImpl.convertToModifiedRemoteApp(RemoteAppsRepository.kt:572)
03-18 14:24:54.983 15992 16019 E AndroidRuntime: 	at com.kieronquinn.app.pixellaunchermods.repositories.RemoteAppsRepositoryImpl.access$convertToModifiedRemoteApp(RemoteAppsRepository.kt:104)
03-18 14:24:54.983 15992 16019 E AndroidRuntime: 	at com.kieronquinn.app.pixellaunchermods.repositories.RemoteAppsRepositoryImpl$getAllModifiedApps$2.invokeSuspend(RemoteAppsRepository.kt:556)
03-18 14:24:54.983 15992 16019 E AndroidRuntime: 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
03-18 14:24:54.983 15992 16019 E AndroidRuntime: 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
03-18 14:24:54.983 15992 16019 E AndroidRuntime: 	at kotlinx.coroutines.internal.LimitedDispatcher.run(LimitedDispatcher.kt:42)
03-18 14:24:54.983 15992 16019 E AndroidRuntime: 	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:95)
03-18 14:24:54.983 15992 16019 E AndroidRuntime: 	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:570)
03-18 14:24:54.983 15992 16019 E AndroidRuntime: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:749)
03-18 14:24:54.983 15992 16019 E AndroidRuntime: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:677)
03-18 14:24:54.983 15992 16019 E AndroidRuntime: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:664)
03-18 14:24:54.983 15992 16019 E AndroidRuntime: 	Suppressed: kotlinx.coroutines.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@a067aae, Dispatchers.Default]
```
[Full log](http://0x0.st/H-OF.log)

**Comment by KieronQuinn** [comment]
This is suggesting there's some corruption in your local database. Clearing data should fix it, though you would lose your settings. 

Given the changes to bitmap handling in the March 2023 update, I wonder if something changed that broke one or more of your icons.

**Comment by matt4682** [comment]
I threw a try-catch inside `RemoteAppsRepositoryImpl.convertToModifiedRemoteApp` and had it log the componentName and return a null `monoIcon` when it encountered a bad icon, and the only thing it was throwing the error on for me was: `com.google.android.apps.walletnfcrel/com.google.commerce.tapandpay.android.cardlist.CardListActivity`.

Thankfully with that try-catch modified app I was able to open it up and create an in-app backup and restore that backup on a fresh install the latest release, which appears to have restored all my settings including the icon for the app it was throwing the error for 🤔

**Comment by KieronQuinn** [comment]
Should be fixed in 2.5
